CodePush.js

```diff
@@ -170,8 +183,31 @@ function setUpTestDependencies(testSdk, providedTestConfig, testNativeBridge) {
   if (testNativeBridge) NativeCodePush = testNativeBridge;
 }
 
+// This function allows only one syncInternal operation to proceed at any given time.
+// Parallel calls to sync() while one is ongoing yields CodePush.SyncStatus.SYNC_IN_PROGRESS.
+const sync = (() => {
+  let syncInProgress = false;
+  const setSyncCompleted = () => { syncInProgress = false; };
+  
+  return (options = {}, syncStatusChangeCallback, downloadProgressCallback) => {
+    if (syncInProgress) {
+      syncStatusChangeCallback(CodePush.SyncStatus.SYNC_IN_PROGRESS);
```

Comment:
Don't you need to check for null here since the callback is optional? Additionally, can we add logging here to indicate the state?